File: typescript-client-sdk/src/sdk/models/operations/postrules.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class PostRulesSecurity extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "security, scheme=true;type=oauth2" })
  codatLogin?: shared.SchemeCodatLogin;

  @SpeakeasyMetadata({ data: "security, scheme=true;type=apiKey;subtype=header" })
  apiKeyAuth?: shared.SchemeApiKeyAuth;
}


export class PostRulesRequest extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "request, media_type=application/json" })
  request?: shared.CodatPublicApiModelsRulesAddRuleModel;

  @SpeakeasyMetadata()
  security: PostRulesSecurity;
}


export class PostRulesResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  codatPublicApiModelsRulesRule?: shared.CodatPublicApiModelsRulesRule;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;
}

File: python-client-sdk/src/sdk/models/operations/post_rules.py
import dataclasses
from typing import Optional
from ..shared import security as shared_security
from ..shared import security as shared_security
from ..shared import codat_public_api_models_rules_addrulemodel as shared_codat_public_api_models_rules_addrulemodel
from ..shared import codat_public_api_models_rules_rule as shared_codat_public_api_models_rules_rule


@dataclasses.dataclass
class PostRulesSecurity:
    api_key_auth: Optional[shared_security.SchemeAPIKeyAuth] = dataclasses.field(default=None, metadata={'security': { 'scheme': True, 'type': 'apiKey', 'sub_type': 'header' }})
    codat_login: Optional[shared_security.SchemeCodatLogin] = dataclasses.field(default=None, metadata={'security': { 'scheme': True, 'type': 'oauth2' }})
    

@dataclasses.dataclass
class PostRulesRequest:
    security: PostRulesSecurity = dataclasses.field()
    request: Optional[shared_codat_public_api_models_rules_addrulemodel.CodatPublicAPIModelsRulesAddRuleModel] = dataclasses.field(default=None, metadata={'request': { 'media_type': 'application/json' }})
    

@dataclasses.dataclass
class PostRulesResponse:
    content_type: str = dataclasses.field()
    status_code: int = dataclasses.field()
    codat_public_api_models_rules_rule: Optional[shared_codat_public_api_models_rules_rule.CodatPublicAPIModelsRulesRule] = dataclasses.field(default=None)
    
